service/tablet_allocator.cc

```diff
@@ -760,20 +760,22 @@ class load_balancer {
         const locator::topology& topo = _tm->get_topology();
         migration_plan plan;
 
-        // Prepare plans for each DC separately and combine them to be executed in parallel.
-        for (auto&& dc : topo.get_datacenters()) {
-            if (_db.get_config().rf_rack_valid_keyspaces()) {
-                for (auto rack : topo.get_datacenter_racks().at(dc) | std::views::keys) {
-                    auto rack_plan = co_await make_plan(dc, rack);
-                    auto level = rack_plan.size() > 0 ? seastar::log_level::info : seastar::log_level::debug;
-                    lblogger.log(level, "Prepared {} migrations in rack {} in DC {}", rack_plan.size(), rack, dc);
-                    plan.merge(std::move(rack_plan));
+        if (!utils::get_local_injector().enter("tablet_migration_bypass")) {
```

Comment:
nit: I would prefer if this injection wouldn't hover so prominently over the main logic. It would be better if we changed make_plan() to have an early return instead.

We keep adding injection points to topology coordinator, and they're getting more and more distracting.

But do it only if there's a new revision.

Reply:
So unit test specifically for tablet_metadata_guard makes a lot of sense.